lib/nats.js

```diff
@@ -1534,29 +1545,33 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
         msg = msg === undefined ? null : msg;
     }
 
+    if (opt_callback && typeof opt_callback !== 'function') {
+        throw (new NatsError(BAD_CALLBACK_MSG, BAD_CALLBACK));
+    }
+
     if (!subject) {
         if (opt_callback) {
             opt_callback(new NatsError(BAD_SUBJECT_MSG, BAD_SUBJECT));
+            return;
         } else {
             throw (new NatsError(BAD_SUBJECT_MSG, BAD_SUBJECT));
         }
     }
     if (typeof msg === 'function') {
-        if (opt_callback || opt_reply) {
+        if (opt_callback) {
             opt_callback(new NatsError(BAD_MSG_MSG, BAD_MSG));
             return;
+        } else {
+            throw(new NatsError(BAD_MSG_MSG, BAD_MSG));
```

Comment:
Normalizing the arguments (checking if some optional arguments have been omitted and then using the last function argument as the callback) is now done at the top.